src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Formatting/CodeActionFormattingTest.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Formatting
+{
+    public class CodeActionFormattingTest : FormattingTestBase
+    {
+        public CodeActionFormattingTest(ITestOutputHelper output)
+            : base(output)
+        {
+        }
+
+        [Fact]
+        public async Task AddDebuggerDisplay()
+        {
+            await RunCodeActionFormattingTestAsync(
+input: @"
+@functions {
+    class Goo$$
+    {
+        
+    }
+}
+",
+codeActionEdits: new []
+{
+    Edit(7, 17, 7, 17, "Diagnostics;\r\n    using System."),
+    Edit(67, 0, 67, 8, ""),
+    Edit(69, 34, 70, 7, "\r\n\r\n        [DebuggerDisplay($\"{{{nameof(GetDebuggerDisplay)}(),nq}}\")]"),
+    Edit(71, 0, 71, 4, "        "),
+    Edit(72, 5, 72, 5, "\r\n            private string GetDebuggerDisplay()\r\n            {"),
+    Edit(73, 0, 73, 0, "                return ToString();\r\n            }\r\n"),
+    Edit(73, 8, 74, 4, "")
+},
+expected: @"
+@functions {
+    [DebuggerDisplay($""{{{nameof(GetDebuggerDisplay)}(),nq}}"")] 
+    class Goo
+    {
+        private string GetDebuggerDisplay()
+        {
+            return ToString();
+        }
+    }
+}
+");
+        }
+
+        [Fact]
+        public async Task GenerateConstructor()
```

Comment:
So this test fails, but the diff shows no difference.. will have to go character by character and work out what is going on...

Also want to add more tests for code blocks and code and functions blocks where the brace is on the next line, but getting the test data for them is a **HUGE PAIN**. I've manually tested all of the permutations though, and they all work.